test/dataset_utils_test.py

```diff
@@ -41,10 +44,56 @@ def to_dict(keys: List[str], df: pd.DataFrame):
 
     Use this to extract the values from a DataFrame for easier comparisons in assert statements.
     """
-    if any(df.index.names):
-        df = df.reset_index()
-    df = df.set_index(keys)
-    return df.to_dict(orient="index", into=NoNanDict)
+    try:
+        if any(df.index.names):
+            df = df.reset_index()
+        df = df.set_index(keys)
+        return df.to_dict(orient="index", into=NoNanDict)
+    except Exception:
+        print(f"Problem with {df}")
```

Comment:
What is the reason for catching and re-raising the error here?

Reply:
added comment